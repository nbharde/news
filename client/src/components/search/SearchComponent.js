import React, { useRef } from "react";
import { useDispatch } from "react-redux";
import { getSearchedNews } from "../../actions/NewsReportsAction";

const SearchComponent = (props) => {
    const inputRef = useRef(null);
    const dispatch = useDispatch();

    const handleSubmit = (e) => {
        e.preventDefault();
        
        const { value } = inputRef.current;
        const request = {
            keyword: value
        }
        
        dispatch(getSearchedNews(request));
    }

    return(
        <div className="searchbox-container" data-testid="searchbox_testId">
            <form autoComplete="off" onSubmit={handleSubmit} data-testid="form_testId">
                <input 
                    type="text" 
                    className="search-box"
                    ref={inputRef}
                    placeholder="Enter Keyword to Search"
                    data-testid="input_testId"
                />
                <button  
                    type="submit" 
                    className="submit-btn"
                    alt="Search"
                    data-testid="submit_btn_testId"
                >Submit</button>
            </form>
        </div>
    );
}

export default SearchComponent