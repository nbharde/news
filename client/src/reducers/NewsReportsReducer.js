const NewsReportsReducer = (newsReports = [], action) => {
    switch (action.type) {
        case 'LIST':
            return action.payload;

        case 'SEARCH':
            return action.payload;
    
        default:
            return newsReports;
    }
}

export default NewsReportsReducer;