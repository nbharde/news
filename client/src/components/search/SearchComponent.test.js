import React from "react";
import { createStore, applyMiddleware, compose } from "redux";
import { Provider } from "react-redux";
import { render, fireEvent, cleanup } from "@testing-library/react";
import thunk from "redux-thunk";
import '@testing-library/jest-dom/extend-expect';

import SearchComponent from "./SearchComponent";
import NewsReportsReducer from "../../reducers/NewsReportsReducer";

afterEach(cleanup);

function renderWithRedux(component, {initialState, store = createStore(NewsReportsReducer, compose(applyMiddleware(thunk)))} = {}) {
    return {
        ...render(<Provider store={store}>{component}</Provider>)
    }
}

describe("Testing SearchComponent", () => {
    it("renders the form with redux", () => {
        const { getByTestId } = renderWithRedux(<SearchComponent />);
        expect(getByTestId("searchbox_testId")).not.toBeNull();
    })

    it("Form can be submited", () => {
        const mockSubmit = jest.fn();
        const { getByTestId } = renderWithRedux(<SearchComponent />);

        fireEvent.change(getByTestId('input_testId'), { target: { value: "aaa" } });
        fireEvent.click(getByTestId('submit_btn_testId')); 

        expect(mockSubmit.mock.calls).toEqual([]); 

    });

    it("matches snapshot", () => {
        const { container } = renderWithRedux(<SearchComponent />);
        expect(container).toMatchSnapshot();
    });
});
