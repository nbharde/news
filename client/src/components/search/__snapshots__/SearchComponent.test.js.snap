// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SearchComponent matches snapshot 1`] = `
<div>
  <div
    class="searchbox-container"
    data-testid="searchbox_testId"
  >
    <form
      autocomplete="off"
      data-testid="form_testId"
    >
      <input
        class="search-box"
        data-testid="input_testId"
        placeholder="Enter Keyword to Search"
        type="text"
      />
      <button
        alt="Search"
        class="submit-btn"
        data-testid="submit_btn_testId"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
